### PR TITLE
Implementation of a new storage backend : filesystem

### PR DESCRIPTION
**Pull request by qgerome**
The default storage used in OpenHEXA is changed to be this new one. It stores the workspaces buckets inside a defined directory in the .env

The .env.dist file has been revamped to better explained the purpose of each variable.

The backend is largely inspired by the filesystem storage of Django.

### CODE REVIEW

**Review by toch — COMMENTED** [review]
First pass done :)

**Review by qgerome — COMMENTED** [review]
Sorry @toch, I didn't understand that I had to submit the review of your review for you to see my comments...

File: hexa/files/backends/fs.py [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+    def create_directory(self, directory_path: str):
+        if self.exists(directory_path):
+            raise self.exceptions.AlreadyExists("Directory already exists")
+        directory = self.path(directory_path)
+
+        try:
+            if self.directory_permissions_mode is not None:
+                # Set the umask because os.makedirs() doesn't apply the "mode"
+                # argument to intermediate-level directories.
+                old_umask = os.umask(0o777 & ~self.directory_permissions_mode)
```
Comment (nazarfil) [inline hexa/files/backends/fs.py:134]: thank you, interesting!

I wonder if  here instead os.mkdirs() - path.mkdir(parents=True) is better to be explicit for the parent folder generation?

**Review by nazarfil — APPROVED** [review]

File: hexa/files/backends/fs.py (hunk 1 above)
Comment (qgerome) [reply hexa/files/backends/fs.py:134]: By looking at the documentation of both calls:
- https://docs.python.org/3/library/pathlib.html#pathlib.Path.mkdir
- https://docs.python.org/3/library/os.html#os.makedirs

Only the `os.makedirs` proposes a way to set the permissions to intermediate-directories

File: hexa/files/backends/fs.py [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+                try:
+                    os.makedirs(
+                        directory, self.directory_permissions_mode, exist_ok=True
+                    )
+                finally:
+                    os.umask(old_umask)
+            else:
+                os.makedirs(directory, exist_ok=True)
+        except FileExistsError:
```
[the remark this answers is not in the document]
Comment (qgerome) [reply hexa/files/backends/fs.py:143]: Indeed

[the remark this answers is not in the document]
Comment (qgerome) [reply hexa/files/backends/fs.py:143]: (this code is largely inspired by Django implementation https://github.com/django/django/blob/1fa84936408b84a6530d56cc1efed83ce6e7160c/django/core/files/storage/filesystem.py#L21)

**Pull request merged**